### PR TITLE
Gun migration and rebalance

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
~~I have removed the barrel length of the AN-94~~, added burst modes to most Rivtech automatic weapons, reduced the weights of the PPSh mags, added "material": [ "steel", "plastic" ], to EBR, migrated the slings and the bullet crossbow to pebble.json, migrated many 40mm launchers to 40mm.json, migrated the bolt action rifles to 300.json, brught the other bolt actions up to code, and migrated the spearguns to fishspear.json.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
Since GitHub doesn't migrate author credit between files, it may be a good idea to either convert ranged.json into a credits file, or add a list of user additions to CONTRIBUTORS

**Comment by illi-kun** [comment]
Git stores history of changes so you always can track down an author of any changes, anything else is excessive, imho.

File: data/json/items/gun/300.json [hunk 1]
```diff
@@ -18,5 +18,57 @@
...
+    "name_plural": "Winchester Model 70",
+    "description": "The Winchester Model 70 is a bolt action sporting rifle.  It has an iconic place in American sporting culture and has been held in high regard by shooters since it was introduced in 1936.",
+    "weight": 4100,
+    "volume": 12,
+    "price": 46000,
+    "bashing": 12,
+    "to_hit": -1,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "skill": "rifle",
```
Comment (mugling) [inline data/json/items/gun/300.json]: `skill` is already defined by `rifle_manual`

File: data/json/items/gun/300.json [hunk 2]
```diff
@@ -18,5 +18,57 @@
...
+    "ammo": "300",
+    "dispersion": 225,
+    "sight_dispersion": 60,
+    "aim_speed": 6,
```
Comment (mugling) [inline data/json/items/gun/300.json]: `aim_speed` is again provided via inheritance

File: data/json/items/gun/300.json [hunk 3]
```diff
@@ -18,5 +18,57 @@
...
+    "weight": 4200,
+    "volume": 12,
+    "price": 259500, "//": "MSRP, believe it or not.",
+    "bashing": 12,
+    "to_hit": -1,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "skill": "rifle",
+    "ammo": "300",
+    "dispersion": 0,
```
Comment (mugling) [inline data/json/items/gun/300.json]: No need to specify defaults `dispersion 0`

File: data/json/items/gun/300.json [hunk 4]
```diff
@@ -18,5 +18,57 @@
...
+    "aim_speed": 6,
+    "recoil": 45,
+    "durability": 8,
+    "clip_size": 3,
+    "reload": 100,
+    "barrel_length": 3,
+    "flags": "RELOAD_ONE"
+  },
+  {
+    "id": "weatherby_5",
```
Comment (mugling) [inline data/json/items/gun/300.json]: alphabetical ordering by `id`

File: data/json/items/gun/40mm.json [hunk 5]
```diff
@@ -1,5 +1,30 @@
...
+    "description": "A simple, home-made grenade launcher.  Basically a tube with a pin firing mechanism to activate the grenade.",
+    "weight": 2267,
+    "volume": 5,
+    "price": 40000,
+    "bashing": 12,
+    "to_hit": -3,
+    "material": [ "steel", "wood" ],
+    "symbol": "(",
+    "color": "brown",
+    "skill": "launcher",
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: Again `skill` is unnecessary (and all further occurrences)

File: data/json/items/gun/40mm.json [hunk 6]
```diff
@@ -16,12 +41,63 @@
...
     "recoil": 270,
     "durability": 8,
     "burst": 10,
     "reload": 400,
     "magazines": [ [ "40mm", [ "belt40mm" ] ] ],
     "flags": "MOUNTED_GUN"
+  },
+  {
+    "id": "rm802",
+    "type": "GUN",
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: Needs to inherit from `launcher_base`

File: data/json/items/gun/40mm.json [hunk 7]
```diff
@@ -16,12 +41,63 @@
...
+    "name": "RM802 grenade launcher",
+    "description": "The Rivtech RM802 pump-action 40mm grenade launcher was designed to be a step ahead of single-shot grenade launchers by allowing sustained heavy firepower.  However, its capacity is fairly limited so it must be used with careful planning.",
+    "weight": 3120,
+    "volume": 6,
+    "price": 2920000, "//": "Either Rivtech or milspec would make it Expensive.  Combined, you could buy a half-decent car for the price.",
+    "bashing": 8,
+    "to_hit": -1,
+    "material": [ "superalloy", "ceramic" ],
+    "symbol": "(",
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: Both `color` and `symbol` are provided by inheritance

File: data/json/items/gun/40mm.json [hunk 8]
```diff
@@ -16,12 +41,63 @@
...
+    "ammo": "40mm",
+    "dispersion": 300,
+    "sight_dispersion": 150,
+    "aim_speed": 0,
+    "recoil": 60,
+    "durability": 7,
+    "burst": 3,
+    "clip_size": 3,
+    "reload": 350,
+    "valid_mod_locations": [[ "accessories", 4 ],[ "grip", 1 ],[ "mechanism", 4 ],[ "rail", 1 ],[ "sights", 1 ],[ "stock", 1 ],[ "underbarrel", 1 ]],
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: This might be best just using the `valid_mod_locations` provided by `launcher_base` from which it should inherit

File: data/json/items/gun/545x39.json [hunk 9]
```diff
@@ -9,19 +9,19 @@
     "price": 390000, "//" : "AKs likewise aren't commercially traded in the US, plus this is newer.",
     "bashing": 11,
     "to_hit": -1,
-    "material": ["steel", "wood"],
+    "material": [ "steel", "wood" ],
```
Comment (mugling) [inline data/json/items/gun/545x39.json]: Never mix unrelated JSON cleanup with behavioral changes or other migrations. This just increases line count.

File: data/json/items/gun/8x40mm.json [hunk 10]
```diff
@@ -74,7 +74,11 @@
     "aim_speed": 4,
     "recoil": 100,
     "durability": 9,
-    "burst": 5,
+    "modes": [
+      [ "DEFAULT", "semi-auto", 1 ],
+      [ "BURST", "2 rd.", 2 ],
```
Comment (mugling) [inline data/json/items/gun/8x40mm.json]: You need to confirm this with @Rivet-the-Zombie

File: data/json/items/gun/300.json (hunk 2 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/300.json]: I just looked at the rifle_manual class in CleverRaven/master and it is 8, not 6.

File: data/json/items/gun/fishspear.json [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    "aim_speed": 0,
+    "recoil": 0,
+    "durability": 8,
+    "clip_size": 1,
+    "reload": 600,
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "grip", 1 ], [ "sights", 1 ], [ "stock", 1 ] ],
+    "flags": [ "STR_RELOAD", "WATERPROOF_GUN", "UNDERWATER_GUN", "NEVER_JAMS" ]
+  },
+  {
+    "id": "doublespeargun",
```
Comment (mugling) [inline data/json/items/gun/fishspear.json]: `copy-from: speargun`

File: data/json/items/gun/fishspear.json [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    "aim_speed": 0,
+    "recoil": 0,
+    "durability": 8,
+    "clip_size": 1,
+    "reload": 700,
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "grip", 1 ], [ "sights", 1 ], [ "stock", 1 ] ],
+    "flags": [ "STR_RELOAD", "WATERPROOF_GUN", "UNDERWATER_GUN", "NEVER_JAMS" ]
+  },
+  {
+    "id": "minispeargun",
```
Comment (mugling) [inline data/json/items/gun/fishspear.json]: `copy-from: speargun`

File: data/json/items/gun/pebble.json [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    "symbol": "(",
+    "color": "green",
+    "skill": "archery",
+    "ammo": "pebble",
+    "ranged_damage": 10,
+    "range": 10,
+    "dispersion": 55,
+    "sight_dispersion": 60,
+    "aim_speed": 6,
+    "recoil": 0,
```
Comment (mugling) [inline data/json/items/gun/pebble.json]: Don't specify default `recoil`

**Comment by mugling** [comment]
This PR doesn't add these guns to any `itemgroups` which should be done at the time of migration. I haven't checked but did your last PR omit that stage also as if so those items will never spawn?

File: data/json/items/gun/40mm.json (hunk 6 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/40mm.json]: What does that do, apart from defining skill and mod hardpoints?

**Comment by BorkBorkGoesTheCode** [comment]
The RM99 was added to pistol_obscure.

File: data/json/items/gun/40mm.json (hunk 6 above)
Comment (Rivet-the-Zombie) [inline data/json/items/gun/40mm.json]: Is copying from the base appropriate here? This weapon's stats are quite different from those of most other launcher-class weapons.

File: data/json/items/gun/8x40mm.json (hunk 10 above)
Comment (Rivet-the-Zombie) [inline data/json/items/gun/8x40mm.json]: I like the extra firing modes.

File: data/json/items/gun/40mm.json (hunk 6 above)
Comment (mugling) [inline data/json/items/gun/40mm.json]: > What does that do, apart from defining skill and mod hardpoints?

Also should define `symbol` and `color`

> Is copying from the base appropriate here? This weapon's stats are quite different from those of most other launcher-class weapons.

The class sets few properties as it is anticipated launchers will vary widely

File: data/json/items/gun/8x40mm.json (hunk 10 above)
Comment (mugling) [inline data/json/items/gun/8x40mm.json]: Good stuff. I'd like to extend this to other appropriate items as it starts making burst fire much more useful

Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/8x40mm.json]: The scout rifle description says it only has a 2-round burst and cannot fire full auto. Can NPCs use a burst-only automatic weapon?

File: data/json/items/gun/40mm.json [hunk 14]
```diff
@@ -85,13 +86,11 @@
     "bashing": 12,
     "to_hit": -3,
     "material": [ "steel", "wood" ],
-    "symbol": "(",
     "color": "dark_gray",
-    "skill": "launcher",
     "ammo": "40mm",
     "dispersion": 300,
     "sight_dispersion": 150,
-    "aim_speed": 0,
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: This should stay as `aim_speed 0`
Currently `aim_speed` is broken and this doesn't fix it

File: data/json/items/gun/pebble.json [hunk 15]
```diff
@@ -59,7 +59,7 @@
     "weight": 254,
     "volume": 1,
     "price": 500,
-    "bashing": 0,
+    "bashing": 1,
```
Comment (mugling) [inline data/json/items/gun/pebble.json]: If it's `bashing 1` it might as well be `bashing 0` and therefore dropped entirely

File: data/json/items/gun/300.json [hunk 16]
```diff
@@ -18,5 +18,54 @@
...
+    "weight": 4200,
+    "volume": 12,
+    "price": 259500, "//": "MSRP, believe it or not.",
+    "bashing": 12,
+    "to_hit": -1,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "ammo": "300",
+    "sight_dispersion": 60,
+    "aim_speed" : 6,
```
Comment (mugling) [inline data/json/items/gun/300.json]: `aim_speed` should not be specified it is inherited from `rifle_manual`

File: data/json/items/gun/300.json [hunk 17]
```diff
@@ -18,5 +18,54 @@
...
+    "recoil": -60,
+    "durability": 8,
+    "clip_size": 3,
+    "reload": 100,
```
Comment (mugling) [inline data/json/items/gun/300.json]: `reload: 100` is the default

File: data/json/items/gun/300.json [hunk 18]
```diff
@@ -18,5 +18,54 @@
...
+    "volume": 12,
+    "price": 46000,
+    "bashing": 12,
+    "to_hit": -1,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "ammo": "300",
+    "dispersion": 225,
+    "sight_dispersion": 60,
+    "aim_speed": 6,
```
Comment (mugling) [inline data/json/items/gun/300.json]: Again `aim_speed` and `reload`

File: data/json/items/gun/40mm.json [hunk 19]
```diff
@@ -1,5 +1,30 @@
...
+    "type": "GUN",
+    "name": "tube 40mm launcher",
+    "description": "A simple, home-made grenade launcher.  Basically a tube with a pin firing mechanism to activate the grenade.",
+    "weight": 2267,
+    "volume": 5,
+    "price": 40000,
+    "bashing": 12,
+    "to_hit": -3,
+    "material": [ "steel", "wood" ],
+    "symbol": "(",
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: Doesn't need `symbol`

File: data/json/items/gun/40mm.json [hunk 20]
```diff
@@ -16,12 +41,62 @@
...
+    "material": [ "superalloy", "ceramic" ],
+    "color": "dark_gray",
+    "ammo": "40mm",
+    "dispersion": 150,
+    "sight_dispersion": 120,
+    "aim_speed": 3,
+    "recoil": 30,
+    "durability": 9,
+    "clip_size": 4,
+    "reload": 100,
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: `reload: 100` is the default

File: data/json/items/gun/40mm.json [hunk 21]
```diff
@@ -16,12 +41,62 @@
...
+    "type": "GUN",
+    "name": "RM802 grenade launcher",
+    "description": "The Rivtech RM802 pump-action 40mm grenade launcher was designed to be a step ahead of single-shot grenade launchers by allowing sustained heavy firepower.  However, its capacity is fairly limited so it must be used with careful planning.",
+    "weight": 3120,
+    "volume": 6,
+    "price": 2920000, "//": "Either Rivtech or milspec would make it Expensive.  Combined, you could buy a half-decent car for the price.",
+    "bashing": 8,
+    "to_hit": -1,
+    "material": [ "superalloy", "ceramic" ],
+    "color": "dark_gray",
```
Comment (mugling) [inline data/json/items/gun/40mm.json]: Can safely default to the base classes `color: gray`

File: data/json/items/gun/40mm.json (hunk 8 above)
Comment (mugling) [inline data/json/items/gun/40mm.json]: Remains outstanding, these slots aren't especially sensible vs the defaults (especially `underbarrel`)

File: data/json/items/gun/545x39.json [hunk 22]
```diff
@@ -36,24 +36,23 @@
...
     "color": "brown",
     "skill": "rifle",
     "ammo": "545x39",
     "dispersion": 35,
     "sight_dispersion": 50,
-    "aim_speed" : 4,
+    "aim_speed": 4,
     "recoil": -10,
     "durability": 7,
-    "barrel_length" : 1,
```
Comment (mugling) [inline data/json/items/gun/545x39.json]: Why the change in `barrel_length`?

File: data/json/items/gun/8x40mm.json (hunk 10 above)
Comment (mugling) [inline data/json/items/gun/8x40mm.json]: Yes

File: data/json/items/gun/fishspear.json [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+    "copy-from": "speargun",
+    "type": "GUN",
+    "name": "mini speargun",
+    "description": "A small underwater fishing implement designed to fire a specialized type of spear using tension in a rubber band.  While highly effective when used underwater, it is nearly useless on dry land.",
+    "weight": 840,
+    "volume": 4,
+    "price": 10000,
+    "color": "yellow",
+    "skill": "pistol",
+    "dispersion": 120,
```
Comment (mugling) [inline data/json/items/gun/fishspear.json]: `dispersion` and `reload` should probably be either `proportional` or `relative` so as to be insensitive to the changes in the base item.

Possibly

```
proportional: {
  dispersion: 1.5
  reload: 0.8
}
```

**Comment by mugling** [comment]
This PR contains behavioral changes, migrations to new JSON format and whitespace changes across 9 files. As a general rule if you need to use the word **and** in your PR title it probably should be multiple PRs.

File: data/json/items/gun/40mm.json (hunk 21 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/40mm.json]: the base class has light_gray, the rm802 is dark_gray

File: data/json/items/gun/40mm.json (hunk 8 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/40mm.json]: Then why is the bayonet compatible with launchers in the first place?

File: data/json/items/gun/40mm.json (hunk 21 above)
Comment (mugling) [inline data/json/items/gun/40mm.json]: Indeed, but some uniformity would actually be useful. It makes sense elsewhere (eg .`brown` for wooden items) but is rather arbitrary here.

File: data/json/items/gun/545x39.json (hunk 22 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/545x39.json]: It has an integral suppressor. It doesn't have much to saw of either due to the gas mechanism.

Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/545x39.json]: Correction, It has a muzzle break, and it doesn't have much to saw off due to the gas mechanism

File: data/json/items/gun/40mm.json (hunk 8 above)
Comment (mugling) [inline data/json/items/gun/40mm.json]: It probably shouldn't be but that's an omission. This entry overrides the class default which shouldn't happen unless there are exceptional circumstances.

File: data/json/items/gun/fishspear.json [hunk 24]
```diff
@@ -37,6 +37,10 @@
     "skill": "pistol",
     "dispersion": 120,
     "reload": 600
+    "relative": {
```
Comment (mugling) [inline data/json/items/gun/fishspear.json]: `proportional` is better than `relative` except when the base item might have a value of zero.
Also JSON error

File: data/json/items/gun/300.json (hunk 16 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/300.json]: I am assuming the preexisting aim speed was intentional. If it was purely game balance, I will remove aim-speed

Comment (mugling) [inline data/json/items/gun/300.json]: `aim_speed` is set randomly throughout the code and currently doesn't do much. I'd like to change this at some point but for now unless there is a reason why this should be different than the other rifles it shouldn't be specified.

File: data/json/items/gun/40mm.json (hunk 8 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/40mm.json]: I am not going to do it in this PR, but how could I separate the underbarrel hardpoint into bayonet and underbarrel hardpoints? It would stop the arguments about flintlock-grenade launcher combinations, and it would allow guns to have bayonets and grenade launchers simultaneously, which is possible irl.

**Comment by mugling** [comment]
This is a massive omnibus PR approaching 400 lines and you keep broadening the scope with each additional commit. There area of behavioral changes hidden in here and some of those lack broad consensus.

I'm sorry but having asked twice previously for you to stay on topic I can't further review this PR.

**Comment by BorkBorkGoesTheCode** [comment]
I have updated the other bolt actions to the rifle-manual standard.
